diff --git a/ios/ExpoImage.podspec b/ios/ExpoImage.podspec
index e74c9d9..89dfa44 100644
--- a/ios/ExpoImage.podspec
+++ b/ios/ExpoImage.podspec
@@ -29,10 +29,8 @@ Pod::Spec.new do |s|
 
   s.dependency 'ExpoModulesCore'
   s.dependency 'SDWebImage', '~> 5.21.0'
-  s.dependency 'SDWebImageAVIFCoder', '~> 0.11.0'
   s.dependency 'SDWebImageSVGCoder', '~> 1.7.0'
   s.dependency 'SDWebImageWebPCoder', '~> 0.14.6'
-  s.dependency 'libavif/libdav1d' unless disable_libdav1d
 
   # Swift/Objective-C compatibility
   s.pod_target_xcconfig = {
diff --git a/ios/ImageModule.swift b/ios/ImageModule.swift
index fc21410..854c95d 100644
--- a/ios/ImageModule.swift
+++ b/ios/ImageModule.swift
@@ -2,7 +2,6 @@
 
 import ExpoModulesCore
 internal import SDWebImage
-internal import SDWebImageAVIFCoder
 internal import SDWebImageSVGCoder
 
 public final class ImageModule: Module {
@@ -352,7 +351,6 @@ public final class ImageModule: Module {
   static func registerCoders() {
     SDImageCodersManager.shared.addCoder(WebPCoder.shared)
     SDImageCodersManager.shared.addCoder(PSDCoder.shared)
-    SDImageCodersManager.shared.addCoder(SDImageAVIFCoder.shared)
     SDImageCodersManager.shared.addCoder(SDImageSVGCoder.shared)
     SDImageCodersManager.shared.addCoder(SDImageHEICCoder.shared)
   }
